src/libsyntax_ext/cfg.rs

```diff
@@ -25,16 +27,39 @@ pub fn expand_cfg<'cx>(cx: &mut ExtCtxt,
                        tts: &[tokenstream::TokenTree])
                        -> Box<dyn base::MacResult + 'static> {
     let sp = sp.apply_mark(cx.current_expansion.mark);
+
+    match parse_cfg(cx, sp, tts) {
```

Comment:
Perhaps we should use `try` here instead of creating a new function?

This could be 

```
let result = try { /* old code */ };
```